core/utils/ip.cc

```diff
@@ -79,7 +79,7 @@ Ipv4Prefix::Ipv4Prefix(const std::string &prefix) {
   } else if (len >= 32) {
     mask = be32_t(0xffffffff);
   } else {
-    mask = be32_t(~((1 << (32 - len)) - 1));
+    mask = be32_t(~((1u << (32 - len)) - 1));
```

Comment:
This is one of those things that should go into that field-of-bits utility header file we were talking about. (meanwhile, fine for now)